### ISSUE TITLE
Terrain: Switching Shadow filter method parameter on the Directional Light component can cause an Editor crash

### ISSUE THREAD

**Issue by jckand-amzn** [o3de/o3de#10400]
**Describe the bug**
Switching Shadow filter method parameter on the Directional Light component can cause an Editor crash. 

**Steps to reproduce**
1. Open a level with existing terrain, or configure simple terrain utilizing the Terrain World Renderer, Terrain World Debugger, and Terrain World level components along with a simple Terrain Layer Spawner with Heightmap assigned.
2. Select the Sun entity in the Atom Default Environment hierarchy.
3. On the Directional Light component, toggle the Shadow filter method parameter.

**Expected behavior**
Shadow filter is updated without issue.

**Actual behavior**
Editor crashes:

```
Call Stack Trace:
14) 00007FF9762EFF27 (KERNELBASE) : UnhandledExceptionFilter
13) 00007FF9789951B0 (ntdll) : memset
12) 00007FF97897C766 (ntdll) : _C_specific_handler
11) 00007FF97899209F (ntdll) : _chkstk
10) 00007FF978941454 (ntdll) : RtlRaiseException
 9) 00007FF978990BCE (ntdll) : KiUserExceptionDispatcher
 8) D:\GitHub\o3de\Gems\Atom\RHI\Code\Source\RHI\DrawListContext.cpp (63) : AZ::RHI::DrawListContext::AddDrawPacket
 7) D:\GitHub\o3de\Gems\Terrain\Code\Source\TerrainRenderer\TerrainMeshManager.cpp (391) : Terrain::TerrainMeshManager::DrawMeshes
 6) D:\GitHub\o3de\Gems\Terrain\Code\Source\TerrainRenderer\TerrainFeatureProcessor.cpp (344) : Terrain::TerrainFeatureProcessor::ProcessSurfaces
 5) D:\GitHub\o3de\Code\Framework\AzCore\AzCore\Task\TaskExecutor.cpp (264) : AZ::Internal::TaskWorker::Run
 4) D:\GitHub\o3de\Code\Framework\AzCore\Platform\Common\WinAPI\AzCore\std\parallel\internal\thread_WinAPI.cpp (38) : AZStd::Internal::thread_run_function
 3) 00007FF976731BB2 (ucrtbase) : configthreadlocale
 2) 00007FF977D57034 (KERNEL32) : BaseThreadInitThunk
 1) 00007FF978942651 (ntdll) : RtlUserThreadStart
```

**Screenshots/Video**
https://user-images.githubusercontent.com/82226555/176235091-1d462876-acb8-454e-aacd-bc7c3c42a094.mp4

**Found in Branch**
development@ 0d1619064c73dcfdee39ab65b6c569700f257e03

**Desktop/Device:**
 - Device: PC
 - OS: Windows 10 21H1
 - CPU: Ryzen 5950x
 - GPU: Nvidia 2070
 - Memory: 64GB

**Comment by invertednormal** [comment]
I'm pretty sure this was fixed with @antonmic's pass changes several weeks ago.

**Comment by jckand-amzn** [comment]
Verified fix on development@ cc6c0964babf3ff5a7d5eed93c1b19852c100012.
